### ISSUE TITLE
The correct way to define the sprocket (and relate a tank's linear speed to the sprocket's rotation speed)

### ISSUE THREAD

**Issue by A1ex-Andro** [Muushy/Sprocket-Feedback#326]
Instead of the sprocket diameter, the user should specify the number of teeth. Then:
1. The tank's speed is calculated from the track pitch and number of teeth.
2. The visual size of the sprocket (as in, how big it looks) is determined by its pitch diameter, which is also calculated from the track pitch and number of teeth.

This is the _correct_ way to relate the speed of the tank to the rotation of the sprocket. If the track pitch is 130 mm and the sprocket has 20 teeth (so that 20 links pass over the sprocket with each rotation), the tank moves 2.6 m with each rotation of the sprocket.

Then, if your engine is at 3000 rpm, your gear ratio is "0.8" and your final drive is "10.0", then your sprocket spins at `3000/(0.8*10) = 375 rpm`. And your tank would move at `375*60*2.6/1000 = 58.5 km/h`.

To summarise, `Speed = RPM * 60 * Pitch * N / (GearRatio * FinalDrive)`.

And this pitch diameter, what do you need it for?
`PitchDiameter = Pitch/sin(180°/N)`
Here's what: Your tracks move along an arc. The pitch diameter defines the arc that passes through the centres of the track pins.
What do you need this for? So that the sprocket appears to be the correct size visually (right now, it does not).
![image](https://user-images.githubusercontent.com/109421040/179356881-37e8ddac-5f80-47c9-98d7-c713745f8987.png)
Questions?
